src/plugins/home/public/application/components/_home.scss

```diff
@@ -71,3 +64,16 @@ $homePageWidth: 1200px;
     margin-top: 0 !important;
   }
 }
+
+.homFooter {
+  // Prevent children from rendering as flex items
+  .euiFlexItem {
+    display: block;
+  }
+
+  // Accounting for no `flush="both"` prop on EuiButtonEmpty
+  .euiButtonEmpty__content {
```

Comment:
These are dangerous selectors. We highly recommend never targeting `.eui` classes directly. Flex items are easy with `EuiFlexItem.className` and there's actually a prop on EuiButtonEmpty to pass a custom class to the content wrapper called `contentProps`.

```tsx
contentProps={{ className: "customClass" }}
```

Reply:
Or, honestly, the better way to handle the margins here is to use the flush property `flush="left"` which removes the padding from both sides of the content, and then adjust the actual margin on the outer button. Like:

```
<EuiButtonEmpty className="homeFooter__button" flush="left" />

.homeFooter__button {
  margin-right: 0;
}
```

Reply:
Hey, @cchaos! Thanks for the comments. I was the one that added these styles. I didn't consider them dangerous selectors because they are nested in a `.homFooter` selector, so they're very specific. Do you still feel it demands an additional class on the flex item itself? Or is this nesting sufficient?

```
.homFooter {
  // Prevent children from rendering as flex items
  .euiFlexItem {
    display: block;
  }

  // Accounting for no `flush="both"` prop on EuiButtonEmpty
  .euiButtonEmpty__content {
    padding-left: 0;
    padding-right: 0;
  }
}
```

As for the styles themselves, the reason I didn't use the `flush` property is something that I outlined in an [EUI GitHub issue](https://github.com/elastic/eui/issues/4012). Because the buttons can either be flush left or right due to media query layout changes, I needed a way to style both the left and right sides as flush (no horizontal padding). 

I considered doing as you described (using `flush="left"` or `flush="right"` and removing the added margining via CSS, but it felt odd to split the singular styling I was attempting to achieve between two methods and files. I figured just simply saying no horizontal padding in the CSS would be much easier for others to find and understand, and also make it easy to remove in the future (assuming a `flush="both"` prop does eventually become available). Thoughts?

Reply:
What we consider dangerous selectors is not just about scope but also about breaks that could happen in upgrades. We don't consider our internal class name changes to be breaking changes but will cause breaks if consumers depend on them. That's why we consciously allow customizing class names on every element where possible. 

The much safer approach is to supply your own class to this content wrapper with `contentProps.className` which will do what you need without targeting the `.eui` specified class.

But the even safer approach is to modify the outer margin because if we decide to change the inner-workings of this component in the future, it will again break your overrides. But the margin will always behave as you intend because it's what's truly causing the spacing.

I see that you've done this override several times in this PR and so it seems that the issue you opened is actually more dire. If that's the case, feel free to reach out to us for help, or even create a PR!

Reply:
Thanks for that explanation. This makes total sense now. I'll go ahead and make these changes!